src/bid/bid.rs

```diff
@@ -88,10 +127,42 @@ impl Bid {
 
     pub fn prove_score_generation(
         &self,
-        composer: &mut StandardComposer,
+        _composer: &mut StandardComposer,
     ) -> Result<Proof, Error> {
-        use crate::score_gen::score::prove_correct_score_gadget;
+        //use crate::score_gen::score::prove_correct_score_gadget;
 
         unimplemented!()
     }
+
+    pub fn random<R>(
+        rng: &mut R,
+        value: &JubJubScalar,
+        secret: &JubJubScalar,
+    ) -> Result<Self, Error>
+    where
+        R: RngCore + CryptoRng,
+    {
+        let bid = Bid {
+            bid_tree_root: BlsScalar::random(rng),
+            consensus_round_seed: BlsScalar::random(rng),
+            latest_consensus_round: BlsScalar::random(rng),
+            latest_consensus_step: BlsScalar::random(rng),
+            elegibility_ts: rng.next_u32(),
+            expiration_ts: rng.next_u32(),
+            prover_id: BlsScalar::default(),
+            score: Score::default(),
+
+            encrypted_data: PoseidonCipher::default(),
+            nonce: BlsScalar::default(),
+            randomness: AffinePoint::identity(),
+
+            secret_k: BlsScalar::random(rng),
+            hashed_secret: BlsScalar::default(),
+            pk: AffinePoint::identity(),
+            c: AffinePoint::default(),
+        };
+
+        let secret = GENERATOR_EXTENDED * secret;
+        bid.init(rng, value, &secret.into())
+    }
```

Comment:
This is not needed and indeed is highly likely to produce a incorrect `Bid` since there's a lot of range checks in respect to elegibility and expiration ts's as well as in respect to the value.

Reply:
Done